### PR TITLE
168 get problem loop on problem page working

### PR DESCRIPTION
**Pull request by aedmondson**
I'm creating this because I used this branch for testing some changes to #174 and found some things that should be relatively easy to fix, so I want to provide documented feedback

### CODE REVIEW

File: ComSemApp/templates/ComSemApp/student/assessment.html [hunk 1]
```diff
@@ -80,6 +97,80 @@ <h2 id="timer-text" class="show-expression"></h2>
...
+    $("#next").removeClass("disabled");
+  }
+  else{
+    $("#next").addClass("disabled");
+  }
+}
+
+function save_in_form(){
+  let attemptFormData = new FormData();
+  attemptFormData.append( 'expression', problems.id);
```
Comment (aedmondson) [inline ComSemApp/templates/ComSemApp/student/assessment.html:150]: problems.id should be problems[curr_problem_index]['formulation']

File: ComSemApp/templates/ComSemApp/student/assessment.html [hunk 2]
```diff
@@ -80,6 +97,80 @@ <h2 id="timer-text" class="show-expression"></h2>
...
+  attemptFormData.append('student', student);
+  attemptFormData.append('audio', audioReformulationBlob);
```
Comment (aedmondson) [inline ComSemApp/templates/ComSemApp/student/assessment.html:152]: audioReformulationBlob should be replaced with audioBlob

File: ComSemApp/templates/ComSemApp/student/assessment.html [hunk 3]
```diff
@@ -80,6 +97,80 @@ <h2 id="timer-text" class="show-expression"></h2>
...
+      recorder.stop();
+      // keeps next problem button disabled if no more problems left
+      if (curr_problem_index < problems.length -1 ){
+        toggle_enabled_next();
+      }
+      
+
+      // updates the progress bar
+      process_response();
+      save_in_form();
```
Comment (aedmondson) [inline ComSemApp/templates/ComSemApp/student/assessment.html:122]: This save_in_form call needs to be dependent on the recorder's dataAvailable event because audioBlob will not be assigned until then.

File: ComSemApp/student/views.py [hunk 4]
```diff
@@ -814,16 +814,17 @@ class SpeakingPracticeAttemptCreateView(SpeakingPracticeView, CreateView):
         Used to process form data served from the SpeakingPracticeView on the frontend.
         Implements the standard Django CreateView
     """
-    model = ReviewAttempt
+    model = SpeakingPracticeAttempt
```
Comment (aedmondson) [inline ComSemApp/student/views.py:817]: Once #174 is merged, you should pull from Placehoder-Pages

File: ComSemApp/templates/ComSemApp/student/assessment.html [hunk 5]
```diff
@@ -80,6 +97,80 @@ <h2 id="timer-text" class="show-expression"></h2>
...
+        type: "POST",
+        url: $("#save_attempt_url").val(),
+        data: attemptFormData,
+        processData: false,
+        contentType: false,
+        headers: {
+           'X-CSRFToken': csrf_token
+        },
+        success: function(response){
+            cs_notification('success', "Expression Saved")
```
Comment (aedmondson) [inline ComSemApp/templates/ComSemApp/student/assessment.html:164]: Just noticed this while doing some usability testing: this says "Expression Saved" when it should say something more along the lines of "Attempt Saved"

**Comment by aedmondson** [comment]
Also, I think I figured out a solution for the next expression button color: when changing its enabled state, also add/remove the classes "btn-success" and "btn-secondary".

File: ComSemApp/templates/ComSemApp/student/assessment.html [hunk 6]
```diff
@@ -80,6 +100,117 @@ <h2 id="timer-text" class="show-expression"></h2>
...
+           'X-CSRFToken': csrf_token
+        },
+        success: function(response){
+            cs_notification('success', "Attempt Saved")
+        },
+        error: function(jqXHR, textStatus, errorThrown){
+            cs_ajax_error(jqXHR, textStatus, errorThrown)
+        },
+  });
+
```
Comment (aedmondson) [inline ComSemApp/templates/ComSemApp/student/assessment.html:179]: This blank line looks a little funky. Not a hill I'm willing to die on, but it might make the code more readable if removed

File: ComSemApp/templates/ComSemApp/student/assessment.html [hunk 7]
```diff
@@ -138,6 +268,12 @@ <h2 id="timer-text" class="show-expression"></h2>
 
     }
   });
+  $("#next").click(() => {
+    console.log("next problem clicked");
+    if(curr_problem_index < problems.length - 1 && !$("#next").hasClass("disabled")){
```
Comment (aedmondson) [inline ComSemApp/templates/ComSemApp/student/assessment.html:273]: This should have parens around the left expressions being evaluated by the && operator to make it a bit more readable: `(curr_problem_index < problems.length - 1) && !$("#next").hasClass("disabled")`

File: ComSemApp/templates/ComSemApp/student/assessment.html [hunk 8]
```diff
@@ -154,24 +290,14 @@ <h2 id="timer-text" class="show-expression"></h2>
...
-    //This was a last test can be deleted
-    recorder.stop();
-    $("#expression-text").html('Problem Complete!');
-  });
-});
+
+//runs the initial reading timer
+timerPromise(sentence)
+
+
```
Comment (aedmondson) [inline ComSemApp/templates/ComSemApp/student/assessment.html:297]: Lots of unnecessary whitespace

File: ComSemApp/templates/ComSemApp/student/assessment.html [hunk 9]
```diff
@@ -154,24 +290,14 @@ <h2 id="timer-text" class="show-expression"></h2>
...
+
 });
 </script>
 
+
```
Comment (aedmondson) [inline ComSemApp/templates/ComSemApp/student/assessment.html:302]: More unnecessary whitespace

**Review by zambroseo — APPROVED** [review]
looks good
